go/oasis-test-runner/scenario/e2e/runtime/txsource.go

```diff
@@ -324,40 +331,90 @@ func (sc *txSourceImpl) manager(env *env.Env, errCh chan error) {
 	livenessTicker := time.NewTicker(sc.livenessCheckInterval)
 	defer livenessTicker.Stop()
 
+	if sc.nodeLongRestartDuration > 0 {
+		sc.Logger.Info("random long node restarts enabled",
+			"start_delay", sc.nodeLongRestartDuration,
+		)
+
+		// Setup long restart routine.
+		go func() {
```

Comment:
Maybe instead of a separate goroutine, you could drive this from the below loop (using a separate ticker or just start the shutdown every so many regular restart ticks) and just trigger the restart action in a separate goroutine.

Reply:
fixed, added a separate ticker